schemas/rum/_common-schema.json

```diff
@@ -204,6 +204,70 @@
       },
       "readOnly": true
     },
+    "os": {
+      "type": "object",
+      "description": "Operating system properties",
+      "required": [
+        "name",
+        "version",
+        "version_major"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Operating system name, ex. Android, iOS",
+          "readOnly": true
+        },
+        "version": {
+          "type": "string",
+          "description": "Full operating system version, ex. 8.1.1",
+          "readOnly": true
+        },
+        "version_major": {
+          "type": "string",
+          "description": "Major operating system version, ex. 8",
+          "readOnly": true
+        }
+      }
+    },
+    "device": {
+      "type": "object",
+      "description": "Device properties",
+      "required": [
+        "type"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Device type info",
+          "enum": [
+            "mobile",
+            "desktop",
+            "tablet",
+            "tv",
+            "gaming_console",
+            "bot",
```

Comment:
Do we have an accurate way to detect those natively? Also on the naming, I might use `gaming_system`, but not sure about this.

Reply:
I guess `gaming_console` may be deducted by having `Xbox`, `Playstation`, etc. in the device name or brand.

Not sure about `Bot` usage (maybe synthetics or case of web crawler?), but we have such type on the backend, so I decided to keep it just in case.

Reply:
Shouldn't we only add them once we need it? I think for now we can only use `mobile`, `tablet` and `tv`.

Reply:
I added them all just in case if Browser SDK start using these properties as well, because this is what I see for example for the rum-app:

![image](https://user-images.githubusercontent.com/4046447/170260833-7e6a7f90-a828-4711-b5f8-9a0bfea507ef.png)

But I'm ok with shrinking this list as well.

Reply:
I agree with @0xnm , these are the values available in the backend and since the browser (also using this rum format) can run on those, we need to be compatible

Reply:
I'm fine with being exhaustive from the start or add it as needed. FYI we don't plan to provide those properties via the Browser SDK.

Reply:
[this](https://cs.github.com/DataDog/logs-backend/blob/3c8cc8db3f0a67ea7b25974423239feaa9efa7d0/domains/event-platform/libs/parsing/src/main/java/com/fsmatic/shared/parse/grok/UserAgentClassifier.java?q=UserAgentClassifier#L5) is the definition of this enum backend side

Reply:
i think you should try to replicate the backend enum values as much as possible